src/node/rpc/member_frontend.h

```diff
@@ -630,15 +808,16 @@ namespace ccf
 
         if (node_operation->is_reading_private_ledger())
         {
-          return make_error(
+          ctx.rpc_ctx->set_error(
             HTTP_STATUS_FORBIDDEN,
             errors::NodeAlreadyRecovering,
             "Node is already recovering private ledger.");
+          return;
         }
 
-        const auto in = params.get<SubmitRecoveryShare::In>();
-        auto raw_recovery_share = crypto::raw_from_b64(in.share);
-        OPENSSL_cleanse(const_cast<char*>(in.share.data()), in.share.size());
+        std::string share = params["share"];
```

Comment:
This seems quite loose compared to what we had before. As per my other comment, if the endpoint was still wrapped with `json_adapter()`, we could keep this as it was.

Reply:
The wrapper can't be used (see above), but we could still disallow additional fields, I agree.